src/core/utils.ts

```diff
@@ -335,14 +335,34 @@ export function multipleValuesQuery(field: string): ExpressionTransformFunction
   return (value: string, operation: string, _options?: any) => {
     const isMultiSelect = isMultiValueExpression(value);
     const valuesArray = getMultipleValuesArray(value);
-    const logicalOperator = getLogicalOperator(operation);
+    const logicalOperator = getConcatOperatorForMultiExpression(operation);
 
     return isMultiSelect
       ? `(${valuesArray.map(val => buildExpression(field, val, operation)).join(` ${logicalOperator} `)})`
       : buildExpression(field, value, operation);
   };
 }
 
+/**
+ * Gets the logical operator for a given query operation when building multi-value expressions
+ * or combining multiple properties.
+ * 
+ * Use Cases:
+ * 1. Multi-value fields: Combines expressions when using multi-value variables
+ *    Example: status = "{active,pending}" → (status = "active" || status = "pending")
+ * 
+ * 2. Multi property fields: Combines expressions when a field corresponds to multiple properties
+ *    Example: source != "sys1" → (system != "sys1" && minionId != "sys1")
+ *  
+ * @param operation The operation to be checked.
+ * @returns The logical operator as a string.
+ */
+export function getConcatOperatorForMultiExpression(operation: string): string {
```

Comment:
This looks query builder related, should consider putting in `query-builder.utils.ts` instead